pkg/operator/k8sutil/pod.go

```diff
@@ -194,25 +206,29 @@ func DeleteDaemonset(clientset kubernetes.Interface, namespace, name string) err
 		_, err := clientset.ExtensionsV1beta1().DaemonSets(namespace).Get(name, metav1.GetOptions{})
 		return err
 	}
-	return deletePodsAndWait(namespace, name, deleteAction, getAction)
+	return deletePodsAndWait(namespace, name, "daemonset", deleteAction, getAction)
 }
 
 // deletePodsAndWait will delete a resource, then wait for it to be purged from the system
-func deletePodsAndWait(namespace, name string,
+func deletePodsAndWait(namespace, name, resourceType string,
```

Comment:
Isn't the function name wrong in point of it deleting a "resource" and not Pods anymore (which it already didn't really do before)?
What about changing it to `deleteResourceAndWait`?

Reply:
agreed, will rename it